include/deal.II/base/vectorization.h

```diff
@@ -68,6 +68,27 @@ namespace std
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  template <typename T>
+  struct NumberType 
```

Comment:
i think the two functions for a general type and `std::complex` could be moved somewhere.
@kronbichler @bangerth : any suggestions?